File: leetcode_python/medium/GrayCode.py
'''
Created on Jan 10, 2015

@author: cksharma
'''
class Solution:
    # @return a list of integers
    def grayCode(self, n):
        ans = list();
        ans.append(0);
        for i in range(n):
            for j in reversed(range(len(ans))):
                ans.append(ans[j] + (1 << i));
        return ans;
if __name__ == '__main__':
    sol = Solution();
    print (sol.grayCode(3));

File: leetcode-cpp/archieve/GrayCode.cpp
#include "header.h"

vector<int> GrayCode::grayCode(int n) {
	vector <int> ans;
	ans.push_back(0);
	for (int i = 0; i < n; i++) {
		for (int j = ans.size() - 1; j >= 0; j--) {
			ans.push_back(ans[j] + (1 << i));
		}
	}
	return ans;
}